### ISSUE TITLE
[BUG] Arithmetic operation fails

### ISSUE THREAD

**Issue by msadeqhe** [hsutter/cppfront#319]
It is the example code:

```
main: () = {
    x := 2*2;
}
```

It should generate the following C++1 code:

```
auto main() -> int{
    auto x {2 * 2}; 
}
```

But C++2 compiler outputs the following error message:

```
error: postfix unary * (dereference) cannot be immediately followed by a (, identifier, or literal - add whitespace before * here if you meant binary * (multiplication)
```

On the other hand, `2+2` works, also `2 * 2` (with white-spaces around `*` operator) works. The reason of this bug seems to be that `*` operator can be both unary postfix and binary. I think it's related to [this issue](https://github.com/hsutter/cppfront/issues/152).

**Comment by JohelEGP** [comment]
According to https://github.com/hsutter/cppfront/wiki/Design-note%3A-Postfix-operators, it should work:
> additionally, as a convenience, cases like `x**y` mean the same as `x* * y` since they can't mean anything else, which is convenient so we can still write things like `a*b` and `a*2` conveniently with the obvious meaning.

**Comment by msadeqhe** [comment]
I think that's because the behaviour is changed after [this commit](https://github.com/Azmah-Bad/cppfront/commit/daaf4f04c11b0276397ad769c9c4d7846124617c).

**Comment by AbhinavK00** [comment]
One way could be to completely rethink the operator symbols (it's a very small space) and come up with something unambigious. Something that'd solve both this issue and #304 .

**Comment by msadeqhe** [comment]
Using `and`, `or` and `not` keywords instead of `&&`, `||` and `!`, seems reasonable to solve the issue with `&` operator, but it doesn't solve the issue with `*` operator.

To solve this issue, if there is unary postfix operators side-by-side a binary operator, and if the last unary postfix operator's symbol is the same as the binary operator's symbol, then there should be a white-space between them, otherwise white-space is not necessary. Also this can be true for unary prefix operators, therefore imagine if C++2 had unary prefix `*` in addition to unary postfix `*` and binary `*`, then:

```
a0 := x***y; // ERROR!
b0 := x* * *y; // OK.

// Here whitespace isn't required.
// := (x--)*(-y);
a1 := x--*-y; // OK.
a2 := 2*2; // It will work too.
```

**Comment by hsutter** [comment]
Thanks! Finally fixed -- see longer note in #989.

The example

```
main: () = {
    x := 2*2;
}
```

obviously should just work, and now it Just Works again.
